src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -934,6 +916,48 @@ async def _genomic_to_tx_segment(
             genomic_ac, genomic_pos, is_seg_start, gene, tx_ac=transcript
         )
 
+    async def _select_optimal_transcript(
+        self, genomic_pos: int, genomic_ac: str, gene: str
+    ) -> str:
+        """Select the optimal transcript given a genomic position, accession, and gene.
+        In this case, optimal refers to the transcript that is the best represenative
+        transcript for this data, given that MANE data does not exist for the
+        gene or the MANE transcript for the gene does not exist in UTA
+
+        :param genomic_pos: Genomic position where the transcript segment starts or ends
+            (inter-residue based)
+        :param genomic_ac: Genomic accession
+        :param gene: Valid, case-sensitive HGNC gene symbol
+        :return A string representing the optimal transcript given the above data
+        """
+        # Attempt to find a coding transcript if a MANE transcript cannot be found
+        transcript = None
+        results = await self.mane_transcript.get_longest_compatible_transcript(
+            start_pos=genomic_pos,
+            end_pos=genomic_pos,
+            start_annotation_layer=AnnotationLayer.GENOMIC,
+            gene=gene,
+        )
+        if results:
+            transcript = results.refseq
+        else:
+            # Run if gene is for a noncoding transcript
+            query = f"""
+                SELECT DISTINCT tx_ac
+                FROM {self.uta_db.schema}.tx_exon_aln_v
+                WHERE hgnc = '{gene}'
+                AND alt_ac = '{genomic_ac}'
+                """  # noqa: S608
+            result = await self.uta_db.execute_query(query)
+
+            if result:
+                transcript = result[0]["tx_ac"]
```

Comment:
I know this logic isn't new in this PR, but not sure if getting the first transcript from UTA makes sense. We may want to follow the transcript priority algorithm similar to what is done in the `ManeTranscript` class (`_get_prioritized_transcripts_from_gene`)

Reply:
I agree. I will try to fix this logic in a new PR